### DISCUSSION TITLE
yuv_420_888 CAMERA2 API image conversion to Mat

### DISCUSSION THREAD

**Discussion by MattiaDurli** [emgucv/emgucv#470] [Q&A]
Hello, the Android Camera2 API live feed returns an Android.Media.Image in yuv_420_888 format.
I've been able to use Emgu on Xamarin Android by converting a Android.Graphics.Bitmap to a Mat, but how can I convert the Android.Media.Image?

Thanks

**Comment by emgucv** [comment]
You can use this:
https://github.com/emgucv/emgucv/blob/master/Emgu.CV.Example/XamarinForms/Core/Android/YUV420Converter.cs
It is part of the Emgu CV Xamarin Android demo program.

**Comment by MattiaDurli** [comment]
Thanks @emgucv, I used it on my App and got weird results, so I tested the Emgu Xamarin Sample, on four devices, a Nexus 5x and Pixel3a (works on both) and a Zebra TC21 and Zebra TC52, where on both I get this:

![Screenshot_20210408-124355](https://user-images.githubusercontent.com/6052847/114015055-9690eb80-9869-11eb-9ba5-dcf03c4ec5cb.png)

It's the same problem I get with my app, but I was testing it only on the Zebra devices at first.
Can you give me any hint on how to solve it?

**Reply by emgucv** [reply]
Not sure about Zebra device. The code has been tested on a few different Android emulator as welll as a few Samsung Galaxy devices. All seems to work.

I would check if the Zebra device supports live view output in yuv_420_888 format. If not, you will need to implement the color coversion for the device's support live view data format. Your output images seem to indicate that the source pixel format is not yuv_420_888.

**Reply by MattiaDurli** [reply]
I checked during debug and imageformat seems to be yuv_420_888 in both cases, or at least so it is reported:

Nexus5x: 
![framenexus](https://user-images.githubusercontent.com/6052847/114058520-d10f7e00-9893-11eb-9e1c-ee69e455213b.png)

Zebra:
![framezebra](https://user-images.githubusercontent.com/6052847/114058671-f69c8780-9893-11eb-8870-e97d79f41ee4.png)

Can you suggest anything else I can do to try to detect the Image format?
